src/kademlia/dht_storage.cpp

```diff
@@ -171,6 +171,22 @@ namespace
 			, immutable_item_comparator(node_ids));
 	}
 
+	constexpr int sample_infohashes_interval_max = 21600;
+	constexpr int infohashes_sample_count_max = 20;
+
+	struct infohashes_sample
+	{
+		std::vector<sha1_hash> samples;
+		time_point created = min_time();
+
+		int count() const { return int(samples.size()); }
+	};
+
+	int clamp(int v, int lo, int hi)
+	{
+		return (v < lo) ? lo : (hi < v) ? hi : v;
```

Comment:
looking at this makes me envision someone getting ``lo`` and ``hi`` mixed up, with very surprising results. should there be an assert ``lo <= hi`` maybe?